### PR TITLE
Feature/change step by entering stepnumber

### LINKED ISSUE #42: Jump-to-step functionality
## Value Proposition
As a user, I want to be able to jump to a specific step in a tutorial so that I don't have to click back through all the previous steps after an interruption.

## Description
capstone-userstory-16

## Acceptance Criteria
- [x] In addition to the current step number, the total number of steps of the tutorial is displayed
- [x] When clicked, the current step number becomes editable and allows a new step number to be entered
- [x] After entering and confirming with Enter or by onBlur, the appropriate step is displayed
- [x] If no new step number is entered or an invalid entry is made, the current step remains displayed

## Tasks
- [x] Create a new branch
- [x] make it work ;)

### PR DESCRIPTION
**Pull request by kai-franke**
In this PR, I made the displayed step number in the TutorialCard editable so you can jump directly to any step number.

### CODE REVIEW

File: components/TutorialCard.js [hunk 1]
```diff
@@ -1,8 +1,40 @@
...
+  function jumpToStep() {
+    if (stepNumberInput !== "") {
+      setCurrentStep(stepNumberInput);
+    } else {
+      setStepNumberInput(step.step);
+    }
+  }
+
+  useEffect(() => {
+    setStepNumberInput(step.step);
```
Comment (anhthu253) [inline components/TutorialCard.js:36]: I m not sure if it s necessary to set the step here. As I understand, every time the step changes, a tutorial card is called, hence the initial set at line 7 should be enough

File: components/TutorialCard.js [hunk 2]
```diff
@@ -14,7 +46,26 @@ export default function TutorialCard({ step }) {
...
+            type="text"
+            inputMode="numeric"
+            min="1"
+            max={totalSteps}
+            pattern="\d*"
+            widthFactor={
+              stepNumberInput.length
+                ? stepNumberInput.length
+                : step.step.toString().length
+            }
```
Comment (anhthu253) [inline components/TutorialCard.js:63]: nice! it shows the obsession of going down to every little detail 🤣

File: components/TutorialCard.js (hunk 1 above)
Comment (kai-franke) [reply components/TutorialCard.js:36]: You are right, @anhthu253 ! I don't need the useEffect here anymore.

Comment (kai-franke) [reply components/TutorialCard.js:36]: I found out what I needed the useEffect for: If I don't enter the step number in the input field but use the prev or next buttons instead, the step number wouldn't update without the useEffect. So it's back in the code. 😁

Comment (anhthu253) [reply components/TutorialCard.js:36]: that what i assumed, that the step value in props is changed when next or prev buttons is clicked. But any action on these button also leads to a tutorialcard being invoked, isnt it? If it s so then the state step number (stepNumberInput ) is also initialized according to the step value in props. That s why i did not understand why stepNumberInput has to be set again inside useEffect
But yeah i can be wrong ;)

**Pull request merged**